src/web3py/contract_tweak.py

```diff
@@ -1,14 +1,102 @@
+from web3.contract.utils import ACCEPTABLE_EMPTY_STRINGS
+from web3.exceptions import BadFunctionCallOutput
+from web3.types import ABI, TxParams, BlockIdentifier, CallOverride, FunctionIdentifier, ABIFunction
+
+
+def call_contract_function(  # pylint: disable=keyword-arg-before-vararg
+    w3: "Web3",
+    address: ChecksumAddress,
+    normalizers: Tuple[Callable[..., Any], ...],
+    function_identifier: FunctionIdentifier,
+    transaction: TxParams,
+    block_id: Optional[BlockIdentifier] = None,
+    contract_abi: Optional[ABI] = None,
+    fn_abi: Optional[ABIFunction] = None,
+    state_override: Optional[CallOverride] = None,
+    ccip_read_enabled: Optional[bool] = None,
+    decode_tuples: Optional[bool] = False,
+    *args: Any,
+    **kwargs: Any,
+) -> Any:
+    """
+    Helper function for interacting with a contract function using the
+    `eth_call` API.
+    """
+    call_transaction = prepare_transaction(
+        address,
+        w3,
+        fn_identifier=function_identifier,
+        contract_abi=contract_abi,
+        fn_abi=fn_abi,
+        transaction=transaction,
+        fn_args=args,
+        fn_kwargs=kwargs,
+    )
+
+    return_data = w3.eth.call(
+        call_transaction,
+        block_identifier=block_id,
+        state_override=state_override,
+        ccip_read_enabled=ccip_read_enabled,
+    )
+
+    if fn_abi is None:
+        fn_abi = find_matching_fn_abi(
+            contract_abi, w3.codec, function_identifier, args, kwargs
+        )
+
+    output_types = get_abi_output_types(fn_abi)
+
+    try:
+        output_data = w3.codec.decode(output_types, return_data)
+    except DecodingError as e:
+        # Provide a more helpful error message than the one provided by
+        # eth-abi-utils
+        is_missing_code_error = (
+            return_data in ACCEPTABLE_EMPTY_STRINGS
+            and w3.eth.get_code(address) in ACCEPTABLE_EMPTY_STRINGS
+        )
+        if is_missing_code_error:
+            msg = (
+                "Could not transact with/call contract function, is contract "
+                "deployed correctly and chain synced?"
+            )
+        else:
+            msg = (
+                f"Could not decode contract function call to {function_identifier} "
+                f"with return data: {str(return_data)}, output_types: {output_types}"
+            )
+        raise BadFunctionCallOutput(msg) from e
+
+    _normalizers = itertools.chain(
+        BASE_RETURN_NORMALIZERS,
+        normalizers,
+    )
+    normalized_data = map_abi_data(_normalizers, output_types, output_data)
+
+    # Fast tweak. Don't need to decode args
+    if decode_tuples and fn_abi["outputs"]:
+        decoded = named_tree(fn_abi["outputs"], normalized_data)
+        normalized_data = recursive_dict_to_namedtuple(decoded)
+
+    if len(normalized_data) == 1:  # pylint: disable=no-else-return
+        return normalized_data[0]
+    else:
+        return normalized_data
```

Comment:
```suggestion
    if len(normalized_data) == 1:
        return normalized_data[0]
    return normalized_data
```

Reply:
This is copy-paste from web3py lib, so I don't want to change here anything, event minor things

Reply:
In future, we just remove this whole code